src/restricted-endpoints/replication/changes/changes.controller.ts

```diff
@@ -31,31 +33,68 @@ export class ChangesController {
    * @param user
    */
   @Get(':db/_changes')
-  changes(
+  async changes(
     @Param('db') db: string,
-    @Query() params,
     @User() user: UserInfo,
-  ): Observable<ChangesResponse> {
-    return this.couchdbService
-      .get<ChangesResponse>(db, '_changes', {
-        ...params,
-        include_docs: true,
-      })
-      .pipe(map((res) => this.filterChanges(res, user)));
+    @Query() params?: ChangesParams,
+  ): Promise<ChangesResponse> {
+    const ability = this.permissionService.getAbilityFor(user);
+    const change = { results: [] } as ChangesResponse;
+    let since = params?.since;
+    while (true) {
```

Comment:
might have been easier to read without the infinite loop, e.g. something like `while(!(!params?.limit || change.pending === 0 || change.results.length >= params.limit))` ?

But it's alright as it is.

Reply:
The problem is that I would need some kind of `repeat ... until` loop that at least runs once. 
That is a bit difficult to achieve with the while loop as it would require a additional variable which would be modified just where the current break is used. So I did not see much benefit.